BUILD.md

```diff
@@ -9,103 +9,136 @@ See also README.md Dependencies section
+```
+mkdir deps/; cd deps
 git clone https://github.com/facebook/folly.git
+```
+__Install glog-dev (includes gflags, libunwind), boost system, double conversion
+if you can find a binary distrubution for your variant of linux:__
+*libjemalloc-dev dependency is optional*
 
-# Getting pre built dependencies: glog-dev (which adds gflags, libunwind)
-# boost system, double conversion
+```
 sudo apt-get install libgoogle-glog-dev libboost-system-dev \
-     libdouble-conversion-dev # and optionally: libjemalloc-dev
-
-# OR
-
-# If double-conversion isn't available:
-
+libdouble-conversion-dev libjemalloc-dev
+```
+*If double-conversion isn't available via apt-get:*
+```
 git clone https://github.com/floitsch/double-conversion.git
 cd double-conversion; cmake . ; make -j && sudo make install
+```
 
+__Otherwise, Build gflags and glog from source__
 
-# Get and install gflags and glog - it's important to build and configure
-# gflags correctly for glog to pick it up and avoid linking errors later
-# or getting a wdt without flags working
+*It's important to build and configure gflags correctly for glog to pick it up
+and avoid linking errors later or getting a wdt without flags working*
+```
 git clone https://github.com/schuhschuh/gflags.git
 mkdir gflags/build
 cd gflags/build
 cmake -D GFLAGS_NAMESPACE=google -D BUILD_SHARED_LIBS=on ..
 make -j && sudo make install
-# Glog
+```
+
+```
 svn checkout http://google-glog.googlecode.com/svn/trunk/ glog
 cd glog
 ./configure # add --with-gflags=whereyouinstalledgflags
 # to avoid ERROR: unknown command line flag 'minloglevel' later
 make -j && sudo make install
+```
 
 
 
 
-# And finally Build wdt itself:
+__Build wdt from source__
+```
 cmake pathtowdtsrcdir -DBUILD_TESTING=on  # skip -D... if you don't want tests
 make -j
 make test
 sudo make install
+```
+# Install on Mac OS X
 
-# Macos
-
-Install Xcode 6 and the command line tools (so
+__Install Xcode 6 and the command line tools__ (so
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++
 exists)
 
-## cmake
-# Until 3.3 is out we need the latest on a Mac to
-# get CXX 11 support: ( http://www.cmake.org/Bug/view.php?id=15355 )
+__Install Cmake__
 
+```
 git clone http://cmake.org/cmake.git
 cd cmake
 ./bootstrap --prefix=/usr --parallel=16
 make -j && sudo make install
+cd ../
+```
+__Install homebrew (http://brew.sh/)__
 
-## Get homebrew at http://brew.sh/
-
-### Glog and gflags and boost
+__Install glog and gflags and boost__
+```
 brew update
 brew install glog gflags boost
-## Double conversion
+```
+__Install Double conversion__
+```
 git clone https://github.com/floitsch/double-conversion.git
-cd double-conversion; cmake . ; make -j && sudo make install
-
-### Get folly source tree
-As above
-
-## finally build wdt itself:
-mkdir wdt_mac
-cd wdt_mac
-# For Xcode:
+cd double-conversion; cmake . ; make -j && sudo make install;
+cd ../
+```
+__Build wdt from source__
+```
+mkdir wdt-mac
+cd wdt-mac
```

Comment:
not 100% sure that having the source below the build dir works well for eclipse - I believe it should be a sister directory
ie I would move that after the 2 git clone and use ../wdt in the cmake line